src/msw/webview_edge.cpp

```diff
@@ -0,0 +1,777 @@
+                        (GetCurrentURL().substr(0, 4) == "file" &&
+                            wxFileName::URLToFileName(GetCurrentURL()).GetFullPath() == uri))
+                    {
+                        // If we are not at the end of the list, then erase everything
+                        // between us and the end before adding the new page
+                        if (m_historyPosition != static_cast<int>(m_historyList.size()) - 1)
+                        {
+                            m_historyList.erase(m_historyList.begin() + m_historyPosition + 1,
+                                m_historyList.end());
+                        }
+                        wxSharedPtr<wxWebViewHistoryItem> item(new wxWebViewHistoryItem(uri, GetCurrentTitle()));
+                        m_historyList.push_back(item);
+                        m_historyPosition++;
+                    }
+                    //Reset as we are done now
+                    m_historyLoadingFromList = false;
+                }
+                return S_OK;
+            })
+        .Get(), &m_navigationCompletedToken);
+    m_webView->add_NewWindowRequested(
+        Callback<IWebView2NewWindowRequestedEventHandler>(
+            [this](IWebView2WebView* WXUNUSED(sender), IWebView2NewWindowRequestedEventArgs* args) -> HRESULT
+            {
+                PWSTR uri;
+                args->get_Uri(&uri);
+                wxString evtURL(uri);
+                AddPendingEvent(wxWebViewEvent(wxEVT_WEBVIEW_NEWWINDOW, GetId(), evtURL, wxString()));
+                args->put_Handled(true);
+                return S_OK;
+            }).Get(), &m_newWindowRequestedToken);
+    LoadURL(m_pendingURL);
+}
+
+void wxWebViewEdge::OnSize(wxSizeEvent& event)
+{
+    UpdateBounds();
+    event.Skip();
+}
+
+void wxWebViewEdge::UpdateBounds()
+{
+    RECT r;
+    wxCopyRectToRECT(GetClientRect(), r);
+    if (m_webView)
+        m_webView->put_Bounds(r);
+}
+
+void wxWebViewEdge::LoadURL(const wxString& url)
+{
+    if (!m_webView)
+    {
+        m_pendingURL = url;
+        return;
+    }
+    if (FAILED(m_webView->Navigate(url.wc_str())))
+        wxLogError("Could not navigate to URL");
+}
+
+void wxWebViewEdge::LoadHistoryItem(wxSharedPtr<wxWebViewHistoryItem> item)
+{
+    int pos = -1;
+    for (unsigned int i = 0; i < m_historyList.size(); i++)
+    {
+        //We compare the actual pointers to find the correct item
+        if (m_historyList[i].get() == item.get())
+            pos = i;
+    }
+    // TODO: wxASSERT_MSG(pos != static_cast<int>(m_impl->m_historyList.size()), "invalid history item");
+    m_historyLoadingFromList = true;
+    LoadURL(item->GetUrl());
+    m_historyPosition = pos;
+}
+
+wxVector<wxSharedPtr<wxWebViewHistoryItem> > wxWebViewEdge::GetBackwardHistory()
+{
+    wxVector<wxSharedPtr<wxWebViewHistoryItem> > backhist;
+    //As we don't have std::copy or an iterator constructor in the wxwidgets
+    //native vector we construct it by hand
```

Comment:
This is not true (any more?) BTW and this code could perfectly well use `wxVector(first, last)` ctor.

Reply:
I would assume that is copy/paste from `webview_ie.cpp` (will take a look)